includes/Checker/Checks/File_Type_Check.php

```diff
@@ -178,6 +182,32 @@ protected function look_for_hidden_files( Check_Result $result, array $files ) {
 		}
 	}
 
+	/**
+	 * Looks for application files and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param array        $files  List of absolute file paths.
+	 */
+	protected function look_for_application_files( Check_Result $result, array $files ) {
+		$application_files = self::filter_files_by_extensions( $files, array( 'a', 'bin', 'bpk', 'deploy', 'dist', 'distz', 'dmg', 'dms', 'DS_Store', 'dump', 'elc', 'exe', 'iso', 'lha', 'lrf', 'lzh', 'o', 'obj', 'phar', 'pkg', 'sh', 'so' ) );
```

Comment:
Nit-pick, but since this is such a long array, better to break it into separate lines. At a minimum, we should put the array on its own line to improve readability.
```suggestion
		$application_files = self::filter_files_by_extensions(
			$files,
			array( 'a', 'bin', 'bpk', 'deploy', 'dist', 'distz', 'dmg', 'dms', 'DS_Store', 'dump', 'elc', 'exe', 'iso', 'lha', 'lrf', 'lzh', 'o', 'obj', 'phar', 'pkg', 'sh', 'so' )
		);
```